### PR TITLE
Create dotnet.yml

### PR DESCRIPTION
**Pull request by vb2ae**
add dotnet build want to start creating packages on github instead of myget

### CODE REVIEW

**Comment by vb2ae** [comment]
@Yinimi could you do a review on this PR?

**Comment by vb2ae** [comment]
@darxis Could you do a review of this pr?

**Comment by Yinimi** [comment]
Man thats a massive PR... ill see if I find some time to look into it

**Comment by darxis** [comment]
Okay, I will take a look but it will take me some time

**Comment by vb2ae** [comment]
Thank you, guys.  Sorry it is so big.  The build started to fail because of outdated platforms.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -6,16 +6,20 @@ on:
...
-
+  schedule:
+    - cron: '26 22 * * 0'
 
 jobs:
   analyse:
     name: Analyse
-    runs-on: windows-2022
-
+    runs-on: windows-latest
```
[the remark this answers is not in the document]
Comment (darxis) [reply .github/workflows/codeql-analysis.yml:15]: I would use windows-2022 everywhere

File: .github/workflows/dotnet.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            -n github 
+            -u ${{ secrets.NUGET_USER }} 
+            -p ${{ secrets.CONSUME_CALIBURN_FEED }} 
+            --store-password-in-clear-text
+            
+    - name: Restore nuget packages
+      run:  msbuild ${{env.caliburn_sln}} -t:restore
+
+    - name: Build app for release
+      run: msbuild ${{env.caliburn_sln}} /t:Build /p:Configuration=Release
```
Comment (darxis) [inline .github/workflows/dotnet.yml:64]: Move `Release` configuration as `env` variable

File: .github/workflows/dotnet.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        fetch-depth: 0 # avoid shallow clone so nbgv can do its work
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: 8.0.x
+    - name: Setup Java SDK
+      uses: actions/setup-java@v4
+      with:
+        distribution: 'microsoft' 
+        java-version: 11
```
Comment (darxis) [inline .github/workflows/dotnet.yml:36]: Use `java-version: 17` or even newer? Like in [codeql-analysis.yml](https://github.com/Caliburn-Micro/Caliburn.Micro/pull/891/files#diff-63bd641104d10e25f141d518a16b22a151d125e12701df2f9e79734b23b90188)

**Review by darxis — COMMENTED** [review]
I finished the review, everything looks fine, I left just some minor comments.

File: .github/workflows/codeql-analysis.yml [hunk 4]
```diff
@@ -24,33 +28,34 @@ jobs:
...
-        dotnet-version: 7.0.100
+        dotnet-version: 8.0.x
       
     - name: Setup Java SDK
-      uses: actions/setup-java@v1
+      uses: actions/setup-java@v4
       with:
-        java-version: 17
+        distribution: 'microsoft' 
+        java-version: 11
```
[the remark this answers is not in the document]
Comment (vb2ae) [reply .github/workflows/codeql-analysis.yml]: done

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (vb2ae) [reply .github/workflows/codeql-analysis.yml]: done

File: .github/workflows/dotnet.yml (hunk 2 above)
Comment (vb2ae) [reply .github/workflows/dotnet.yml]: done

File: .github/workflows/dotnet.yml (hunk 3 above)
Comment (vb2ae) [reply .github/workflows/dotnet.yml]: done

**Comment by vb2ae** [comment]
@darxis thanks

**Review by darxis — APPROVED** [review]
Looks good, thanks

**Pull request merged**